llvm/lib/Transforms/Yk/StackMaps.cpp

```diff
@@ -62,14 +69,19 @@ class YkStackmaps : public ModulePass {
             // can't tell if an indirect call is an intrinsic at compile time,
             // emit a stackmap in those cases too.
 
-            if (!CI.isIndirectCall() &&
-                (CI.getCalledFunction()->isIntrinsic() ||
-                 (CI.getCalledFunction()->isDeclaration() &&
-                  (!CI.getCalledFunction()->getName().startswith(
-                       "__yk_promote") &&
-                   CI.getCalledFunction()->getName() != YK_NEW_CONTROL_POINT))))
-              continue;
-
+            if (!CI.isIndirectCall()) {
+              if (CI.getCalledFunction()->isIntrinsic()) {
+                // This also skips the control point which is called via a
+                // patchpoint intrinsic, which already emits a stackmap.
+                continue;
+              }
+
+              if (CI.getCalledFunction()->isDeclaration() &&
+                  !CI.getCalledFunction()->getName().startswith(
+                      "__yk_promote")) {
```

Comment:
I think that `__yk_promote` is a constant somewhere that we can use instead of hard-coding it here?

Reply:
It wasn't a defined constant before, so I left it like this.

Ideally it'd be a constant, but that's not what we are doing here.

Reply:
OK.